### DISCUSSION TITLE
Share uses of Send to Custom Command feature

### DISCUSSION THREAD

**Discussion by advice2020** [geany/geany#4062] [General]
In a recent post I made about looking for a way to be able to remove text from lines without removing the lines itself, one of the solutions mentioned was to use the following with the Geany feature "Send To Custom Command"
`sed 's/^.*$//'`

I was just wondering if anyone had any other uses of this feature that they would like to share?
Tricky because everyone probably has different uses for this program, but was just curious what others might be using this feature for, that maybe other users can benefit from knowing about.
 
Thank You

**Comment by eht16** [comment]
Here is mine:
![Screenshot_2024-11-21_23-25-22](https://github.com/user-attachments/assets/6844a4ed-93e3-415d-9c1a-34919aef68c2)

As config:
```ini
custom_commands=sort;sort -u;python3 -mjson.tool;geany_camel_case.py;black --line-length 120 --skip-string-normalization -;sqlformat --reindent -;base64 -w 0;base64 -d;
custom_commands_labels=;;JSON PP;CamelCase;Black Formatter;SQLFormat;Base64 Encode;Base64 Decode;
```

The `geany_camel_case.py` script converts snake_case to lowerCamelCase and vice versa. It can be found on https://github.com/eht16/dotconf/blob/master/user/bin/geany_camel_case.py.

If this gets more, it might be good to build a list in the wiki similar to the snippets (https://wiki.geany.org/snippets/start).

**Comment by advice2020** [comment]
@eht16 
Ahh thank you for sharing these.
Also, so much information in this post you made I do not even know where to begin :)
Literally have recently been trying to figure out some things with the "sort" command and did not even think about using this with this feature, those are some good ones.

I was not aware/thinking about this having an actual config for this, as you mentioned "As config"
I have to look into this a little more.

I was not even that aware of that Geany wiki page, tons of stuff.

I agree, that is great idea, if enough items are shared here, that could be a good idea to give this feature its own wiki page.
But even the 8 you mentioned and the 1 I mentioned provided by b4n is a good start, I think there is also an example found in the manual as well.

Camelcase makes me think of something I was looking to mention in the past, case based tools.  I might make a separate post about this.
I have seen a few posts made by others regarding this, some I can see make use of your Camelcase script so thank you for creating it.

Thank you again for sharing these examples plus all the other information I was not aware of as well.
